tools/xdmod-ondemand-export/xdmod_ondemand_export/xdmod_ondemand_export.py

```diff
@@ -0,0 +1,526 @@
+        self.__new_last_line = line.strip()
+        self.__new_last_request_time = (
+            self.__entry_time_field_to_str(
+                entry.request_time_fields,
+                'timestamp',
+            )
+        )
+        return encoded_line
+
+    def __entry_time_field_to_str(self, time_fields, key):
+        return (
+            time_fields[key].strftime(self.__time_format)
+            if key in time_fields
+            and time_fields[key] is not None
+            else '-'
+        )
+
+    def __convert_to_combined_logformat(self, entry):
+        return (
+            self.__entry_value_to_str(entry.remote_host)
+            + ' ' + self.__entry_value_to_str(entry.remote_logname)
+            + ' ' + self.__entry_value_to_str(entry.remote_user)
+            + ' ' + self.__entry_time_field_to_str(
+                entry.request_time_fields,
+                'timestamp',
+            )
+            + ' "' + self.__entry_value_to_str(entry.request_line)
+            + '" ' + self.__entry_value_to_str(entry.final_status)
+            + ' ' + self.__entry_value_to_str(entry.bytes_sent)
+            + ' "' + self.__entry_headers_in_to_str(entry, 'Referer')
+            + '" "' + self.__entry_headers_in_to_str(entry, 'User-Agent')
+            + '"\n'
+        )
+
+    def __entry_value_to_str(self, value):
+        return '-' if value is None else str(value)
+
+    def __entry_headers_in_to_str(self, entry, key):
+        return (
+            entry.headers_in[key]
+            if hasattr(entry, 'headers_in')
+            and key in entry.headers_in
+            and entry.headers_in[key] is not None
+            else '-'
+        )
+
+    def __write_conf(self):
+        self.__logger.debug('Writing values back to configuration file.')
+        comment_header = """\
+# This is a configuration file used by xdmod_ondemand_export.py.
+#
+# Set the value of "url" in the [destination] section to specify where the logs
+# will be POSTed.
+#
+# Set the values in the [logs] section to tell the script where to find logs
+# and how to parse them.
+#
+# The values in the [prev_run] section will be overwritten in-place in this
+# file by the script when it runs. The value of "last_line" will be set to the
+# last line in the last successfully-POSTed file. The value of
+# "last_request_time" will be set to the %t value of "last_line". In subsequent
+# runs, lines will only be processed if their %t value is greater than or equal
+# to "last_request_time" (but if a line identical to "last_line" is
+# encountered, it will be ignored). Thus, to control which lines are processed,
+# you can set the value of "last_request_time" to be before the %t value of the
+# first line you want processed; just make sure the value of
+# "last_request_time" is in the format {0}, e.g.,
+# [31/May/2023:23:59:59 -0500].
+
+"""
+        comment_header = comment_header.format(self.__time_format)
+        with open(self.__args.conf_path, 'w') as conf_file:
```

Comment:
I don't think it is a good idea to overwrite the configuration file - if something goes wrong with this (non-atomic) operation then the whole system will be hosed.

I would suggest either:
- runtime metadata be stored in its own file seperate from the admin editable config file
- at a minumum write the new version to a temporary file and then use a unix mv to atomically update the configuration file. If you do it this way then there is no chance of an error in the script corrupting the config file.